state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -397,31 +355,36 @@ pub mod pallet {
 		///
 		/// - [BroadcastAttemptTimeout](Event::BroadcastAttemptTimeout)
 		fn on_initialize(block_number: BlockNumberFor<T>) -> frame_support::weights::Weight {
-			// NB: We don't want broadcasts that timeout to ever expire. We will keep retrying
-			// forever. It's possible that the reason for timeout could be something like a chain
-			// halt on the external chain. If the signature is valid then we expect it to succeed
-			// eventually. For outlying, unknown unknowns, these can be something governance can
-			// handle if absolutely necessary (though it likely never will be).
+			// We treat a time out here as a Broadcast Failure. This is handled the same way - the
+			// current broadcaster is reported as Failed to broadcast, and a new broadcaster is
+			// nominated. If there are no more broadcaster available, then the broadcast is aborted.
 			let expiries = Timeouts::<T, I>::take(block_number);
+			let pending_broadcasts = PendingBroadcasts::<T, I>::get();
+			let delayed_retries = DelayedBroadcastRetryQueue::<T, I>::take(block_number);
```

Comment:
I tried to unify the two - but it as slightly more complicated than a very simple change - I will leave this to another PR.
We should get this next PR into the same release as this PR to avoid adding another round of migration for this pallet.

Reply:
100% can leave to another PR